src/oci.c

```diff
@@ -394,31 +425,72 @@ cc_oci_kill (struct cc_oci_config *config,
 		if (! cc_oci_state_file_create (config, state->create_time)) {
 			g_critical ("failed to recreate state file");
 		}
-		return false;
+		goto error;
 	}
 #endif //UNIT_TESTING
 
-	if (signum == SIGKILL) {
-		/* cc-shim is not able to catch the exit code of a process
-		 * finished with SIGKILL signal hence cc-shim MUST receive
-		 * this signal too
-		 */
-		if (kill (state->pid, signum) < 0 && errno != ESRCH) {
-			g_critical ("failed to stop cc-shim %u: %s",
-					(unsigned)state->pid,
-					strerror (errno));
-			return false;
+	int tries = 1;
+	/* termination signals */
+	const int term_signals[] = {
+		SIGTERM, SIGINT, SIGQUIT, SIGKILL, SIGHUP, -1
+	};
+	/* check if signum is a termination signal */
+	for (const int *i = term_signals; i && *i > 0; ++i) {
+		if (signum == *i) {
+			/* if signum is a termination signal we have to wait
+			 * a little bit to know if cc-shim finished
+			 */
+			tries = MAX_TRIES_KILL;
+			break;
 		}
 	}
 
-	last_status = config->state.status;
+	/* cc-shim is not able to catch exit code of a process
+	 * finished with SIGKILL hence cc-shim MUST receive
+	 * this signal too.
+	 * If cc-shim is not running DO NOT report errors
+	 */
+	if (kill (state->pid, signum) < 0 && errno != ESRCH) {
+		g_critical ("failed to stop cc-shim %u: %s",
+				(unsigned)state->pid,
+				strerror (errno));
+		goto error;
+	}
 
-	config->state.status = OCI_STATUS_STOPPED;
+	for (int i=0; i<tries; i++) {
```

Comment:
I agree, but cc-oci-runtime is not the reaper of cc-shim so cc-oci-runtime has to decide if it must kill the container or not